biz.aQute.junit.platform/resources/META-INF/services/org.junit.platform.engine.TestEngine

```diff
@@ -0,0 +1 @@
+aQute.junit.bundle.engine.BundleEngine
```

Comment:
Maybe we should dog food and use the new Bnd [`aQute.bnd.annotation.spi.*`](https://github.com/bndtools/bnd/tree/master/biz.aQute.bnd.annotation/src/aQute/bnd/annotation/spi) annotations to define the META-INF/services file.

Reply:
I had a quick look... this would work I think but would also create a dependency on the service loader registrar service, which I've managed to avoid.

Reply:
So how does the service loader work in Bndtools? Perhaps @rotty3000 could advise here.

Reply:
I manually set the thread context classloader to the current bundle before invoking the JUnit Platform Launcher. This is basically what SPIFly does when it instruments a provider bundle. This works because I know exactly which TestEngine I wasn't to load, and which bundle it's in.

Reply:
https://github.com/kriegfrj/bnd/blob/5b60222a7db9b008d116ff36b9e390d7719d61c7/biz.aQute.junit.platform/src/aQute/junit/platform/Activator.java#L116-L125

I would have preferred to instantiate BundleEngine directly and pass it directly to the JUnit Platform launcher, but this requires junit-platform-launcher 1.3.0 or higher. The Eclipse baseline we're building against is 1.0.0.

Reply:
What version of Eclipse introduced junit-platform-launcher 1.3.0?

Reply:
> This is basically what SPIFly does when it instruments a provider bundle.

Sorry, that should have been "consumer bundle". Provider bundles are not instrumented.

Reply:
I'm not sure when 1.3.0 became part of Eclipse. What I have works though so I wouldn't want to upgrade it just for this, even though it would make the code neater.